### DISCUSSION TITLE
Lots of MQTT messages flying around

### DISCUSSION THREAD

**Discussion by witek1308** [taligentx/dscKeybusInterface#50] [dscKeybusInterface Discussion]
Hi,
I monitored traffic in my network and observed a constant storm of MQTT messages from NodeMCU to broker - around 20/s. I have number of mqtt devices at home and they all together generate only small fraction of the traffic comparing to NodeMCU with HomeAssisant-MQTT example installed.
I can't understand whether they are sent constantly by PC1616 (and therefore PC1616 is to blame) or rather sketch can be improved to limit it and send it only when anything changes. 

After some time count of subscribe request messages is about 40k. It is a mix of Subscribe Request, Ack and constant Publish from the NodeMCU. I can't find clientId in it and don't know where is it coming from and not sure how Mosquitto treats it - as new subscribe and therefore can collapse after sometime when memory for clients reaches full or else.

Does anyone else experience it too? Can this be somehow limited to only situation when change happens?

**Comment by witek1308** [comment]
Update - I realised this is because all my setting were assuming QoS=1 whereas code doesn't handle them this way and therefore all messages were retransmitted endlessly

**Reply by armandvanderwalt** [reply]
What exactly did you have to change here?
